src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxFactoryTests.cs

```diff
@@ -119,6 +120,31 @@ public void TestFreeFormTokenFactory_SpecialTokenKinds()
         }
 
         [Fact]
+        public void TestFreeFormTokenFactoryExceptionMessage_SpecialTokenKinds()
+        {
+            // Make sure the exception message displays the correct namespace/class to use
+            List<SyntaxKind> syntaxKinds = new List<SyntaxKind>()
+            {
+                SyntaxKind.IdentifierToken,
+                SyntaxKind.CharacterLiteralToken,
+                SyntaxKind.NumericLiteralToken
+            };
+            foreach (var kind in syntaxKinds)
+            {
+                try
+                {
+                    SyntaxFactory.Token(default(SyntaxTriviaList), kind, "text", "valueText", default(SyntaxTriviaList));
+                    AssertEx.Fail("Should have thrown");
+                    return;
+                }
+                catch (ArgumentException e)
+                {
+                    Assert.Contains("Microsoft.CodeAnalysis.CSharp.SyntaxFactory", e.Message);
```

Comment:
I'm not too sure about verifying the entire message, as I would just be creating a single-language version of a resource string in a different place, that then needs to be maintained for any change to the text.  I think it would make sense to get the full type name of SyntaxFactory to dynamically set the test value though, making the unit test re-factor proof.  Right now, if SyntaxFactory was re-named or moved to a different namespace, the unit test wouldn't catch that the error message is wrong again.

I will also add a unit test that calls the 3 proper API methods, to validate those work.